### ISSUE TITLE
fin_cases breaks on intervals of Fin 

### ISSUE THREAD

**Issue by shortc1rcuit** [leanprover-community/mathlib4#14652]
`fin_cases` fails on some intervals over the type `Fin (k + 2)`. For example:
```
import Mathlib.Tactic

example (k : ℕ) (x : Fin (k + 2)) (hx : x ∈ Finset.Iic 1) : True := by
  fin_cases hx
```
and
```
import Mathlib.Tactic

example (k : ℕ) (x : Fin (k + 2)) (hx : x ∈ Finset.Icc 0 1) : True := by
  fin_cases hx
```
should each create two cases, (`x = ⊥` and `x = 1` in the first example, `x = 0` and `x = 1` in the second). Instead they both return the error "(kernel) declaration has metavariables '_example'". More examples of this can be found in [this thread](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/fin_cases.20on.20Finset.2EIoc) on the Lean Zulip.

**Comment by plp127** [comment]
@b-reinke I believe that is a bug in `simp`, not `fin_cases`, since if you replace `simp` with `sorry` you get no kernel error.

**Comment by plp127** [comment]
@shortc1rcuit I can no longer reproduce the examples you give here on the web server.

**Comment by b-reinke** [comment]
@plp127, yes, you are right, see [#mathlib4 > ✔ fin_cases kernel error? @ 💬](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/.E2.9C.94.20fin_cases.20kernel.20error.3F/near/504837174) and the [lean4 issue](https://github.com/leanprover/lean4/issues/7444) for `simp`

**Comment by bryangingechen** [comment]
It seems this may have been fixed at some point. Closing for now, but feel free to open a new issue if there's still a problem here.

**Comment by plp127** [comment]
This code still drops metavariables.
```lean-4
import Mathlib.Tactic

variable (k : ℕ)

example (x : Finset.Iic (1 : Fin (k + 1))) : True := by
  fin_cases x
```

**Comment by plp127** [comment]
It shouldn't do that
